### ISSUE TITLE
3.3.4 unzipped file is empty

### ISSUE THREAD

**Issue by yiiim** [brendan-duncan/archive#225]
```dart

import 'dart:io';

import 'package:archive/archive_io.dart';

void main(List<String> args) async {
  var zipPath = "./test.zip";
  var encoder = ZipFileEncoder();
  encoder.open(zipPath);
  await encoder.addDirectory(Directory("./test"));
  encoder.close();

  final inputStream = InputFileStream("./test.zip");
  final archive = ZipDecoder().decodeBuffer(inputStream);
  for (var element in archive.files) {
    if (element.isFile) {
      final outputStream = OutputFileStream('./test_out/${element.name}');
      element.writeContent(outputStream, freeMemory: false);
      await outputStream.close();
    }
  }
  await inputStream.close();
}

```
<img width="964" alt="image" src="https://user-images.githubusercontent.com/12887926/203222059-1edc43ee-96b8-4911-bb16-a1d13bb91093.png">

can work in 3.3.1 

unzipped file is empty in 3.3.4

**Comment by NightDrivers** [comment]
I encountered the same problem, but it also not work in 3.3.1. I thought something wrong for macOS 12.6.1

**Comment by nasko25** [comment]
I am experiencing the same issue.
If I call `file.content` before `file.writeContent(outputStream)` it unzips and saves the file as expected.

**Comment by juliansteenbakker** [comment]
Issue is introduced in version 3.3.3. 
version 3.3.2 seems to be working fine.

**Comment by brendan-duncan** [comment]
It turns out I had been sitting on the fix for this for a week. Sorry about that, work got busy. I published it in 3.3.5.

**Comment by yiiim** [comment]
@brendan-duncan  thanks
